src/QMCDrivers/DMC/DMCBatched.cpp

```diff
@@ -33,13 +33,25 @@
 #include "TauParams.hpp"
 #include "WalkerLogManager.h"
 #include "CPU/math.hpp"
+#include "QMCHamiltonians/NonLocalTOperator.h"
 
 namespace qmcplusplus
 {
 using std::placeholders::_1;
 using WP       = WalkerProperties::Indexes;
 using PsiValue = TrialWaveFunction::PsiValue;
 
+class DMCBatched::DMCContextForSteps : public ContextForSteps
+{
+public:
+  DMCContextForSteps(RandomBase<FullPrecRealType>& random_gen, const NonLocalTOperator& non_local_ops)
+      : ContextForSteps(random_gen), non_local_ops(non_local_ops)
+  {}
+
+  ///non local operator
+  NonLocalTOperator non_local_ops;
```

Comment:
Can this be protected or private?

Reply:
ContextForSteps and its derived class are intended to be just a struct. Marking it private and adding an access function are unnecessary.